OpenEXR/IlmImf/ImfTiledMisc.cpp

```diff
@@ -361,28 +361,42 @@ getTiledChunkOffsetTableSize(const Header& header)
     //
     // Calculate lineOffsetSize.
     //
-    int lineOffsetSize = 0;
+    Int64 lineOffsetSize = 0;
     const TileDescription &desc = header.tileDescription();
     switch (desc.mode)
     {
         case ONE_LEVEL:
         case MIPMAP_LEVELS:
             for (int i = 0; i < numXLevels; i++)
-                lineOffsetSize += numXTiles[i] * numYTiles[i];
-            break;
+            {
+                lineOffsetSize += Int64(numXTiles[i]) * Int64(numYTiles[i]);
+                if ( lineOffsetSize > INT_MAX )
+                {
+                    throw IEX_NAMESPACE::LogicExc("Maximum number of tiles exceeded");
+                }
+            }
+           break;
         case RIPMAP_LEVELS:
             for (int i = 0; i < numXLevels; i++)
+            {
                 for (int j = 0; j < numYLevels; j++)
-                    lineOffsetSize += numXTiles[i] * numYTiles[j];
-            break;
+                {
+                     lineOffsetSize += Int64(numXTiles[i]) * Int64(numYTiles[j]);
+                     if ( lineOffsetSize > INT_MAX )
+                     {
+                        throw IEX_NAMESPACE::LogicExc("Maximum number of tiles exceeded");
+                     }
+                }
+            }
+           break;
         case NUM_LEVELMODES :
             throw IEX_NAMESPACE::LogicExc("Bad level mode getting chunk offset table size");
     }
 
     delete[] numXTiles;
     delete[] numYTiles;
 
-    return lineOffsetSize;
+    return int(lineOffsetSize);
```

Comment:
how about static_caste<int>(

Reply:
Since these recent issues have involved the intricacies of type conversion, I'd recommend adopting the convention of static_cast<size_t>(x) instead of size_t(x) or (size_t) x. It's more verbose, but it states more definitively what's going on. Also, I agree with @meshula, use std::numeric_limits instead of INT_MAX.